tests/src/model/pages/pods-details-page.ts

```diff
@@ -0,0 +1,91 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { Locator, Page } from 'playwright';
+import { BasePage } from './base-page';
+import { PodsPage } from './pods-page';
+import { waitUntil, waitWhile } from '../../utility/wait';
+import { PodState } from '../core/states';
+
+export class PodDetailsPage extends BasePage {
+  readonly labelName: Locator;
+  readonly heading: Locator;
+  readonly closeLink: Locator;
+  readonly backToPodsLink: Locator;
+  readonly podName: string;
+
+  static readonly SUMMARY_TAB = 'Summary';
+  static readonly LOGS_TAB = 'Logs';
+  static readonly INSPECT_TAB = 'Inspect';
+  static readonly KUBE_TAB = 'Kube';
+
+  constructor(page: Page, name: string) {
+    super(page);
+    this.podName = name;
+    this.labelName = page.getByLabel('name').and(page.getByText('Pod Details'));
+    this.heading = page.getByRole('heading', { name: this.podName });
+    this.closeLink = page.getByRole('link', { name: 'Close Details' });
+    this.backToPodsLink = page.getByRole('link', { name: 'Go back to Pods' });
+  }
+
+  async activateTab(tabName: string) {
+    const tabItem = this.page.getByRole('link', { name: tabName });
+    await tabItem.waitFor({ state: 'visible', timeout: 2000 });
+    await tabItem.click();
+  }
+
+  async getState(): Promise<string> {
+    for (const state in PodState) {
+      const stateDiv = this.getPage().getByTitle(state.toUpperCase(), { exact: true });
+      try {
+        await stateDiv.waitFor({ state: 'visible', timeout: 200 });
+        return state.toUpperCase();
+      } catch (error) {
+        // go to next state
+      }
+    }
+    return PodState.Unknown;
+  }
+
+  async stopPod(failIfStopped = false): Promise<void> {
+    try {
+      await waitUntil(async () => (await this.getState()) === PodState.Degraded, 3000, 900);
+      const stopButton = this.page.getByRole('button').and(this.page.getByLabel('Stop Pod'));
+      await stopButton.waitFor({ state: 'visible', timeout: 2000 });
+      await stopButton.click();
+    } catch (error) {
+      if (failIfStopped) {
+        throw Error(`Pod is not running, its state is: ${await this.getState()}, stop button not available: ${error}`);
+      }
+    }
+  }
+
+  async deletePod(timeout: number): Promise<PodsPage> {
+    const deleteButton = this.page.getByRole('button').and(this.page.getByLabel('Delete Pod'));
+    await deleteButton.click();
+    await waitWhile(
+      async () => await this.heading.isVisible(),
+      timeout,
+      1000,
```

Comment:
I think this timeout will not be enough, please change to at least 5000.

Reply:
Agreed, increasing to 5s